lib/Net/Async/Blockchain/BTC.pm

```diff
@@ -95,41 +99,19 @@ L<Net::Async::Blockchain::Client::ZMQ>
 
 =cut
 
-sub zmq_client : method {
+sub zmq_client {
     my ($self) = @_;
     return $self->{zmq_client} //= do {
-        $self->new_zmq_client();
+        $self->add_child(
+            $self->{zmq_client} = Net::Async::Blockchain::Client::ZMQ->new(
+                endpoint    => $self->subscription_url,
+                timeout     => $self->subscription_timeout,
+                msg_timeout => $self->subscription_msg_timeout,
+            ));
+        $self->{zmq_client};
```

Comment:
```suggestion
            my $zmq_client = Net::Async::Blockchain::Client::ZMQ->new(
                endpoint    => $self->subscription_url,
                timeout     => $self->subscription_timeout,
                msg_timeout => $self->subscription_msg_timeout,
            ));
        $zmq_client;
```

Not sure, but I think this will cause a memory lake by incrementing the $self instance reference counts
Maybe I'm wrong, but for safety, I suggest this solution :smile: